From 2c4d3c61aa33f3169e7923ea3433c71e3df0952f Mon Sep 17 00:00:00 2001
From: Frank Tang <ftang@chromium.org>
Date: Mon, 7 Nov 2022 22:08:05 +0000
Subject: [PATCH 1/2] Prepare for ICU72 landing

The date format may output U+202F Narrow SPACE between time
and AM/PM. So switch to use UTF8 and not restrict to ASCII which
will break when the output is not only ASCII.

Bug: 1371208
Change-Id: I05f9daadfba5c45d618f15ae23340da373f2f36a
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/4007076
Reviewed-by: Daniel Rubery <drubery@chromium.org>
Commit-Queue: Frank Tang <ftang@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1068343}

(cherry picked from commit ce2ffed84b95e2b7c068d6f066d86cd034212425)
---
 .../safe_browsing/content/browser/web_ui/safe_browsing_ui.cc    | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/components/safe_browsing/content/browser/web_ui/safe_browsing_ui.cc
+++ b/components/safe_browsing/content/browser/web_ui/safe_browsing_ui.cc
@@ -456,7 +456,7 @@
 std::string UserReadableTimeFromMillisSinceEpoch(int64_t time_in_milliseconds) {
   base::Time time =
       base::Time::UnixEpoch() + base::Milliseconds(time_in_milliseconds);
-  return base::UTF16ToASCII(base::TimeFormatShortDateAndTime(time));
+  return base::UTF16ToUTF8(base::TimeFormatShortDateAndTime(time));
 }
 
 void AddStoreInfo(const DatabaseManagerInfo::DatabaseInfo::StoreInfo store_info,
--- a/third_party/blink/renderer/platform/text/character_property_data.h
+++ b/third_party/blink/renderer/platform/text/character_property_data.h
@@ -107,7 +107,7 @@
     0x1F3FB, 0x1F3FF,
 
     // Transport
-    0x1F6DD, 0x1F6DF,
+    0x1F6DC, 0x1F6DF,
 
     // Colored circles and squares for use with emoji.
     0x1F7E0, 0x1F7EB,
@@ -118,25 +118,25 @@
     0x1F900, 0x1F90F,
     // ZIPPER-MOUTH FACE...SIGN OF THE HORNS
     0x1F910, 0x1F918, 0x1F919, 0x1F97F, 0x1F980, 0x1F9BF, 0x1F9C0, 0x1F9FF,
-    // Clothing
-    0x1FA70, 0x1FA74,
-    // Medical symbols
-    0x1FA78, 0x1FA7C,
+    // Clothing, heart and Medical symbols
+    0x1FA70, 0x1FA7C,
     // Toys and sport symbols
-    0x1FA80, 0x1FA86,
+    0x1FA80, 0x1FA88,
     // Miscellaneous objects
-    0x1FA90, 0x1FAAC,
     // Animals and nature
-    0x1FAB0, 0x1FABA,
+    0x1FA90, 0x1FABD,
+    // Animal
     // Body parts
     // People
-    0x1FAC0, 0x1FAC5,
+    0x1FABF, 0x1FAC5,
+    // animal-mammal
+    0x1FACE, 0x1FACF,
     // Food and drink
-    0x1FAD0, 0x1FAD9,
+    0x1FAD0, 0x1FADB,
     // Face
-    0x1FAE0, 0x1FAE7,
+    0x1FAE0, 0x1FAE8,
     // Hand
-    0x1FAF0, 0x1FAF6,
+    0x1FAF0, 0x1FAF8,
     };
 
 // https://html.spec.whatwg.org/C/#prod-potentialcustomelementname
